### PR TITLE
Improve JSON empty row fix to use less memory

### PR DESCRIPTION
**Pull request by andygrove**
Closes https://github.com/NVIDIA/spark-rapids/issues/8542

Change representation of empty JSON row to just specify a single column instead of specifying all columns in the schema.

This reduces memory overhead without affecting the functionality.

### CODE REVIEW

**Comment by andygrove** [comment]
build

**Review by razajafri — COMMENTED** [review]
I don't understand how this will help improve memory usage. Can you please explain?

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala [hunk 1]
```diff
@@ -36,24 +36,14 @@ case class GpuJsonToStructs(
...
-      case array: ArrayType => s"[${constructEmptyRow(array.elementType)}]"
-      case _ => "{}"
-    }
+      case struct: StructType if struct.fields.nonEmpty =>
+        s"""{"${escapeFieldName(struct.head.name)}":null}"""
+      case other =>
+        throw new IllegalArgumentException(s"$other is not supported as a top level type")    }
   }
 
+  private def escapeFieldName(fieldName: String): String = fieldName.replace("\"", "\\\"")
```
Comment (gerashegalov) [inline sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala:45]: Is [org.apache.commons.text.StringEscapeUtils](https://commons.apache.org/proper/commons-text/javadocs/api-release/org/apache/commons/text/StringEscapeUtils.html) escapeJson/Java of any help?

**Review by gerashegalov — APPROVED** [review]
LGTM

**Comment by andygrove** [comment]
> I don't understand how this will help improve memory usage. Can you please explain?

Sure. This comment in the issue explains it well: https://github.com/NVIDIA/spark-rapids/issues/8542#issuecomment-1584667039

**Review by revans2 — DISMISSED** [review]
If we really want to go overboard we could select the smallest name in the schema, but this looks great as is. It is likely a very minor improvement as it is anyways.

File: sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala (hunk 1 above)
Comment (andygrove) [reply sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala]: Thanks. I  pushed a commit to use this.

**Review by revans2 — APPROVED** [review]

**Comment by revans2** [comment]
build

**Pull request merged**